### PR TITLE
Setheader default

### PR DESCRIPTION
**Pull request by koin612**
I'm also thinking to change the config from

`<setHeader name="Cache-Control" value="no-store" ifAbsent="true" />`

to

`<setHeader name="Cache-Control" valueIfAbsent="no-store" />`

the second looks better to me but I feel like the first one fits better into current membrane config.

### CODE REVIEW

File: core/src/main/java/com/predic8/membrane/core/interceptor/misc/SetHeaderInterceptor.java [hunk 1]
```diff
@@ -72,7 +83,12 @@ public Outcome handleResponse(Exchange exc) throws Exception {
     }
 
     private Outcome handleMessage(Exchange exchange, Message msg) {
-        msg.getHeader().setValue(name, evaluateExpression(new ExchangeEvaluationContext(exchange, msg).getStandardEvaluationContext()));
+        var msgContainsHeader = Arrays.stream(msg.getHeader().getAllHeaderFields()).anyMatch(headerField -> headerField.getHeaderName().equals(name));
```
Comment (predic8) [inline core/src/main/java/com/predic8/membrane/core/interceptor/misc/SetHeaderInterceptor.java]: add hasField() to Header(s)
make sure case is ignored
add tests

File: core/src/main/java/com/predic8/membrane/core/interceptor/misc/SetHeaderInterceptor.java [hunk 2]
```diff
@@ -72,7 +83,10 @@ public Outcome handleResponse(Exchange exc) throws Exception {
     }
 
     private Outcome handleMessage(Exchange exchange, Message msg) {
-        msg.getHeader().setValue(name, evaluateExpression(new ExchangeEvaluationContext(exchange, msg).getStandardEvaluationContext()));
+        if (!ifAbsent || !msg.getHeader().contains(name)) {
```
Comment (predic8) [inline core/src/main/java/com/predic8/membrane/core/interceptor/misc/SetHeaderInterceptor.java]: Can you reverse the condition and make it as a guard? It is hard to understand unterwise.

File: core/src/main/java/com/predic8/membrane/core/http/Header.java [hunk 3]
```diff
@@ -191,6 +191,11 @@ public HeaderField[] getAllHeaderFields() {
 		return fields.toArray(new HeaderField[0]);
 	}
 
+	public boolean contains(String header) {
+		return fields.stream()
+				.anyMatch(headerField -> headerField.getHeaderName().equals(header));
```
Comment (predic8) [inline core/src/main/java/com/predic8/membrane/core/http/Header.java]: Do we need ignorecase?
Add tests to HeaderTest

**Review by predic8 — CHANGES_REQUESTED** [review]

File: core/src/main/java/com/predic8/membrane/core/http/Header.java (hunk 3 above)
Comment (koin612) [reply core/src/main/java/com/predic8/membrane/core/http/Header.java]: Header -> HeaderName class already handles equals ignoring case in it's equals override

File: core/src/main/java/com/predic8/membrane/core/interceptor/misc/SetHeaderInterceptor.java (hunk 2 above)
Comment (koin612) [reply core/src/main/java/com/predic8/membrane/core/interceptor/misc/SetHeaderInterceptor.java]: but then its like

```java
if (bool) return CONTINUE;
setStuff()...
return CONTINUE;
```

and the function always and multiple times returns CONTINUE

Comment (koin612) [reply core/src/main/java/com/predic8/membrane/core/interceptor/misc/SetHeaderInterceptor.java]: changed to guard

**Review by predic8 — APPROVED** [review]

**Review by predic8 — APPROVED** [review]

**Pull request merged**